tests/test/builtin_impls/unsize.rs

```diff
@@ -193,6 +215,29 @@ fn ty_to_dyn_unsizing() {
             "Unique; substitution [], lifetime constraints []"
         }
 
+        // TypeOutlives test
+        // FIXME: this should create a constraint 'a: 'a, but currently
```

Comment:
Well, techincally no. We know `'a: 'a`, so we shouldn't generate this constraint. Maybe the right test here is 
```
forall<'a, 'b> {
  FooLifetime<'a>: Unsize<dyn Principal + Auto + 'b>
}
```
which should be `No possible solution` but might be `Unique` here?